lib/index.ts

```diff
@@ -123,8 +123,13 @@ const mapSchemaTypeToFieldSchema = ({
   const meta: any = {};
 
   for (const metaProp of props) {
-    if (value && value[metaProp] != null) {
-      meta[metaProp] = value[metaProp];
+    const metaValue = value[metaProp];
+    const isValidEnum = metaProps === 'enum' ? Array.isArray(metaValue) : true;
+    const isValidRequired = metaProps === 'required' ? Array.isArray(metaValue) || typeof metaValue === 'boolean' : true;
+    const isValidDescription = metaProps === 'description' ? typeof metaValue === 'string' : true;
```

Comment:
@JulianKingman Wouldn't a more succinct way to address validity of meta props be to check the non-existence of { type }? This is how mongoose requires overrides of meta props in schema definitions.
```
enum: '' // meta
enum: { type: something } // your field
```